Components/MineSharp.Protocol/Packets/Handlers/HandshakePacketHandler.cs

```diff
@@ -1,42 +1,35 @@
-﻿using MineSharp.Data.Protocol;
+﻿using MineSharp.Core.Common.Protocol;
+using MineSharp.Data.Protocol;
 using MineSharp.Protocol.Exceptions;
-using MineSharp.Protocol.Packets.Serverbound.Handshaking;
 
 namespace MineSharp.Protocol.Packets.Handlers;
 
-internal class HandshakePacketHandler : IPacketHandler
+internal sealed class HandshakePacketHandler : GameStatePacketHandler
 {
     private readonly MinecraftClient client;
 
     public HandshakePacketHandler(MinecraftClient client)
+        : base(GameState.Handshaking)
     {
         this.client = client;
     }
 
-    public Task HandleIncoming(IPacket packet)
+    public override Task HandleIncoming(IPacket packet)
     {
         return Task.CompletedTask;
     }
 
-    public Task HandleOutgoing(IPacket packet)
+    public override Task HandleOutgoing(IPacket packet)
     {
         return packet switch
         {
-            HandshakePacket handshake => HandleHandshake(handshake),
             _ => throw new UnexpectedPacketException(
```

Comment:
Yes the `HandleOutgoing` method is dead now. I added a small comment in the base class.
Probably because of race conditions I do not get this exception.
But I will remove that method now.